Agms 554 (#747)
aerogear/mobile-docs: modules/ROOT/pages/_partials/showcase-apps/service-setup.adoc

[#setup]
== Setting up Mobile Services to Demonstrate Showcase Apps

// tag::excludeDownstream[]
.Prerequisites

* You are running OpenShift with {product-name} as described in xref:getting-started.adoc[Setting up AeroGear {product-name} on OpenShift].

// end::excludeDownstream[]

.Procedure

// tag::excludeDownstream[]
. To demonstrate {metrics-service}, xref:mobile-metrics.adoc#provisioning-metrics-service[Provision the {metrics-service} service] For more information about features, see xref:mobile-metrics.adoc[{metrics-service} Service]
// end::excludeDownstream[]

. To demonstrate {keycloak-service}:

// tag::excludeDownstream[]

> .. xref:{identity-management.adoc}#provisioning-the-identity-management-service[Provision the {keycloak-service} service].

Reviewer: Use of xref upstream?

// end::excludeDownstream[]

.. link:{idm-guide-link}#configuring-the-service[Set up the {keycloak-service} service].

.. Create a new user to authenticate using the link:{idm-docs-link}[{keycloak-service} documentation].

.. Set credentials for the created user. Follow the steps link:{idm-docs-link}[documentation].

+
For more information about features, see link:{idm-guide-link}[{keycloak-service} Service]

// tag::excludeDownstream[]

> . To demonstrate {unifiedpush-service}xref:push-notifications.adoc#provisioning-push-notifications-service[Provision the {unifiedpush-service} service]. For more information about features, see xref:push-notifications.adoc[{unifiedpush-service} Service]

Reviewer: Use of xref upstream?

// end::excludeDownstream[]

// tag::excludeUpstream[]

> . To demonstrate {unifiedpush-service} the see link:{push-guide-link}[{unifiedpush-service} Service]

Reviewer: ```suggestion
. To demonstrate {unifiedpush-service}, see link:{push-guide-link}[{unifiedpush-service} Service]
```

--
{unifiedpush-service} Service is not supported on Xamarin.
--
////
====

[#binding]
== Binding the Showcase {mobile-client} to {product-name}

// tag::excludeDownstream[]

> . xref:mobile-metrics.adoc#binding-a-mobile-client-with-the-metrics-service[Bind the {mobile-client} to the {metrics-service} service] if you want to demonstrate the {metrics-service} service.

Reviewer: Use of xref upstream?

// end::excludeDownstream[]

. link:{idm-guide-link}#binding-a-mobile-client-with-the-identity-management-service[Bind the {mobile-client} to the {keycloak-service} service] if you want to demonstrate authentication.
. link:{push-guide-link}#binding-a-mobile-client-with-the-push-notifications-service[Bind the {mobile-client} to the {unifiedpush-service} service] if you want to demonstrate {unifiedpush-service}.
+
NOTE: This step also creates a variant as described in link:{push-guide-link}[ {unifiedpush-service} Service] terminology.

. link:{sync-guide-link}#sync-server-binding[Bind the {mobile-client} to the {sync-service} app] if you want to demonstrate {sync-service}.

//[[downloading-the-mobile-services-configuration-file]]
= Downloading the Mobile Services Configuration File

. Open your {mobile-client} in Mobile Developer Console.
. Copy the `mobile-services.json` configuration to your clipboard.
 Save the contents of the clipboard to a new file called `mobile-services.json`.
+

> NOTE: The mobile-services.json file is the link between the mobile services running on OpenShift and the mobile app you are developing. This file provides all required configuration to initialise the various SDKs and get them hooked up/connected to the back-end services.

Reviewer: ```suggestion
NOTE: The mobile-services.json file is the link between the mobile services running on OpenShift and the mobile app you are developing. The mobile-services.json file provides all required configuration to initialise the various SDKs and connect them to the back-end services.
```

+

> . Copy the contents of the `mobile-services.json` file and paste them into the `mobile-services.json` located in `src`.

Reviewer: ```suggestion
. Copy the contents of the `mobile-services.json` file into the `mobile-services.json` file located in `src`.
```
